feature/experimental/gribi/ate_tests/dut_daemon_failure/dut_daemon_failure_test.go

```diff
@@ -0,0 +1,378 @@
+	getResponse, err := clientA.Fluent(t).Get().WithNetworkInstance(*deviations.DefaultNetworkInstance).WithAFT(fluent.IPv4).Send()
+	if err != nil {
+		t.Errorf("Cannot Get: %v", err)
+	}
+	entries := getResponse.GetEntry()
+	var found bool = false
+	for _, entry := range entries {
+		v := entry.Entry.(*grps.AFTEntry_Ipv4)
+		if prefix := v.Ipv4.GetPrefix(); prefix != "" {
+			if prefix == ateDstNetCIDR {
+				found = true
+				t.Logf("Found route to %s in gRIBI Get response", ateDstNetCIDR)
+				break
+			}
+		}
+	}
+	if !found {
+		t.Errorf("Route to %s NOT found in gRIBI Get response", ateDstNetCIDR)
+	}
+}
+
+// gNOIKillProcess kills a daemon on the DUT, given its name and pid.
+func gNOIKillProcess(ctx context.Context, t *testing.T, args *testArgs, pName string, pID uint32) {
+	gnoiClient := args.dut.RawAPIs().GNOI().Default(t)
+	killRequest := &gnps.KillProcessRequest{Name: pName, Pid: pID, Signal: gnps.KillProcessRequest_SIGNAL_KILL, Restart: true}
+	killResponse, err := gnoiClient.System().KillProcess(context.Background(), killRequest)
+	t.Logf("Got kill process response: %v\n\n", killResponse)
+	if err != nil {
+		t.Fatalf("Failed to execute gNOI Kill Process, error received: %v", err)
+	}
+}
+
+func TestDUTDaemonFailure(t *testing.T) {
+
+	start := time.Now()
+	dut := ondatra.DUT(t, "dut")
+	ctx := context.Background()
+
+	// Configure the DUT.
+	t.Logf("Configure DUT")
+	configureDUT(t, dut)
+
+	// Configure the ATE.
+	t.Logf("Configure ATE")
+	ate := ondatra.ATE(t, "ate")
+	top := configureATE(t, ate)
+	top.Push(t).StartProtocols(t)
+
+	t.Logf("Time check: %s", time.Since(start))
+
+	args := &testArgs{
+		ctx: ctx,
+		dut: dut,
+		ate: ate,
+		top: top,
+	}
+
+	t.Run("SetupGribiConnection", func(t *testing.T) {
+
+		// Set parameters for gRIBI client clientA.
+		// Set Persistence to true.
+		clientA := &gribi.Client{
+			DUT:                  dut,
+			FibACK:               false,
+			Persistence:          true,
+			InitialElectionIDLow: 10,
+		}
+
+		t.Log("Establish gRIBI client connection")
+		if err := clientA.Start(t); err != nil {
+			t.Fatalf("gRIBI Connection for clientA could not be established")
+		}
+
+		t.Run("AddRoute", func(t *testing.T) {
+			t.Logf("Add gRIBI route to %s and verify through Telemetry and Traffic", ateDstNetCIDR)
+			addRoute(ctx, t, args, clientA)
+
+			t.Run("VerifyAFT", func(t *testing.T) {
+				verifyAFT(ctx, t, args)
+			})
+
+			t.Run("VerifyTraffic", func(t *testing.T) {
+				verifyTraffic(ctx, t, args)
+			})
+
+		})
+
+		t.Logf("Time check: %s", time.Since(start))
+
+	})
+
+	t.Run("KillGribiDaemon", func(t *testing.T) {
+
+		// Find the PID of gRIBI Daemon.
+		var pName string = "unknown"
+		var pId uint64
+		t.Run("FindGribiDaemonPid", func(t *testing.T) {
+			switch dut.Vendor() {
+			case ondatra.JUNIPER:
+				pName = "rpd"
+			case ondatra.CISCO:
+				pName = "unknown-cisco" //TODO
+			case ondatra.ARISTA:
+				pName = "unknown-arista" //TODO
```

Comment:
@sambhavjain-204 what is the process name of the gRIBI daemon on Arista? This is for TE-8.1 to kill gRIBI process.

Reply:
it is `Gribi`